src/test/groovy/org/prebid/server/functional/tests/EidsSpec.groovy

```diff
@@ -0,0 +1,200 @@
+        given: "PBs with openx bidder"
+        def pbsService = pbsServiceFactory.getService(
+                ["adapters.openx.enabled" : "true",
+                 "adapters.openx.endpoint": "$networkServiceContainer.rootUri/auction".toString()])
+
+        and: "Default bid request with eidpremissions and openx bidder"
+        def eidSource = PBSUtils.randomString
+        def openxEid = new Eid(source: eidSource, uids: [new Uid(id: PBSUtils.randomString)])
+        def genericEid = new Eid(source: PBSUtils.randomString, uids: [new Uid(id: PBSUtils.randomString)])
+        def eids = [openxEid, genericEid]
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            user = new User(eids: eids)
+            imp[0].ext.prebid.bidder.openx = Openx.defaultOpenx
+            ext.prebid.data = new ExtRequestPrebidData(eidpermissions: [new EidPermission(source: eidSource, bidders: [OPENX])])
+        }
+
+        when: "PBS processes auction request"
+        pbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain two bidder request"
+        def bidderRequests = bidder.getBidderRequests(bidRequest.id)
+        assert bidderRequests.size() == 2
+
+        and: "Generic bidder should contain one eid"
+        assert bidderRequests.user.eids.sort { it.size() }[0].sort().containsAll([genericEid].sort())
+
+        and: "Openx bidder should contain two eids"
+        assert bidderRequests.user.eids.sort { it.size() }[1].sort().containsAll([genericEid, openxEid].sort())
+    }
+
+    def "PBs eid permissions for non existing source should not stop auction"() {
+        given: "PBs with openx bidder"
+        def pbsService = pbsServiceFactory.getService(
+                ["adapters.openx.enabled" : "true",
+                 "adapters.openx.endpoint": "$networkServiceContainer.rootUri/auction".toString()])
+
+        and: "Default bid request with eidpremissions and openx bidder"
+        def firstEid = new Eid(source: PBSUtils.randomString, uids: [new Uid(id: PBSUtils.randomString)])
+        def secondEid = new Eid(source: PBSUtils.randomString, uids: [new Uid(id: PBSUtils.randomString)])
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            user = new User(eids: [firstEid, secondEid])
+            imp[0].ext.prebid.bidder.openx = Openx.defaultOpenx
+            ext.prebid.data = new ExtRequestPrebidData(
+                    eidpermissions: [new EidPermission(source: PBSUtils.randomString, bidders: [OPENX])])
+        }
+
+        when: "PBS processes auction request"
+        pbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain two bidder request"
+        def bidderRequests = bidder.getBidderRequests(bidRequest.id)
+        assert bidderRequests.size() == 2
+
+        and: "Openx and Generic bidder should contain two eid"
+        bidderRequests.user.eids.each {
+            assert it.containsAll([secondEid, firstEid])
+        }
```

Comment:
Please add 
`assert is.size() == 2` 
or change: containsAll all to `assert it.sort() == [secondEid, firstEid].sort()`
to make sure we don't have extra elements